cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1131,150 +1366,211 @@ aggregate_reader_metadata::select_row_groups(
   std::optional<std::reference_wrapper<ast::expression const>> filter,
   rmm::cuda_stream_view stream) const
 {
-  // Compute total number of input row groups
-  size_type total_row_groups = [&]() {
-    if (not row_group_indices.empty()) {
-      size_t const total_row_groups =
-        std::accumulate(row_group_indices.begin(),
-                        row_group_indices.end(),
-                        size_t{0},
-                        [](size_t sum, auto const& pfm) { return sum + pfm.size(); });
-
-      // Check if we have less than 2B total row groups.
-      CUDF_EXPECTS(total_row_groups <= std::numeric_limits<cudf::size_type>::max(),
-                   "Total number of row groups exceed the size_type's limit");
-      return static_cast<size_type>(total_row_groups);
-    } else {
-      return num_row_groups;
-    }
+  // Input row group indices must be either empty or equal to the number of data sources
+  CUDF_EXPECTS(row_group_indices.empty() or row_group_indices.size() == per_file_metadata.size(),
+               "Encountered unexpected number of input row group indices");
+
+  // Maximum number of rows in all row groups
+  auto const max_num_rows = get_num_rows();
+
+  // Compute the row bounds if no row group indices are provided, else return zeros
+  auto [rows_to_skip, rows_to_read] = [&]() {
+    if (not row_group_indices.empty()) { return std::pair<int64_t, size_type>{}; }
+    auto const from_opts =
+      cudf::io::detail::skip_rows_num_rows_from_options(skip_rows_opt, num_rows_opt, max_num_rows);
+    CUDF_EXPECTS(from_opts.second <= static_cast<int64_t>(std::numeric_limits<size_type>::max()),
```

Comment:
I'm surprised linter does not complain about this. looks like a good place to use std::cmp

Reply:
I think it may be because we aren't using these variables anywhere like the other place where we were comparing them with the current src idx.